packages/edit-site/src/components/block-editor/navigation-editor.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { store as blockEditorStore } from '@wordpress/block-editor';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+const FOCUSABLE_ENTITIES = [ 'wp_template_part', 'wp_navigation' ];
+
+/**
+ * Wrapper for Navigation focus mode specific block editor.
+ *
+ * @param {Object} options
+ * @param {string} options.templateType the template type of the current view
+ */
+export function useNavigationFocusMode( { templateType } ) {
```

Comment:
Can we move all this logic within the NavigationProvider. That hook is a bit weird in the sense that it's not clear what it does.

Reply:
We can. I was thinking of leaving it as a hook but colocating it alongside the `NavigationProvider` in the same file.

I could also rename it for clarity.

Reply:
I removed the hook.